fastly/errors.go

```diff
@@ -327,11 +328,23 @@ func NewHTTPError(resp *http.Response) *HTTPError {
 		return &e
 	}
 
+	// Save a copy of the body as it's read/decoded.
+	// If decoding fails, it can then be used (via addDecodeErr)
+	// to create a generic error containing the body's read contents.
+	var bodyCp bytes.Buffer
+	body := io.TeeReader(resp.Body, &bodyCp)
+	addDecodeErr := func() {
```

Comment:
In all the places this is called, we actually have a non-nil `err` variable.  I wonder if we could include it in the error object somehow (instead of a generic `Undefined error`).

Reply:
Thanks for the justification.

Reply:
This is possible, and so I wired it up locally and tested.

As an example, imagine that `api.fastly.com` returns an `Fastly error: unknown domain.` plaintext error response body. Users would see something like the following:

```
502 - Bad Gateway

    Title:  Undefined error
    Detail: Fastly error: unknown domain.
    Meta:   map[json_decode_error: invalid character 'F' looking for beginning of value]
```

I'm afraid that the decode error could be conflated with the actual error, leading to confusion.